lsp-clients.el

```diff
@@ -806,6 +805,67 @@ responsiveness at the cost of possible stability issues."
                   :server-id 'emmy-lua
                   :notification-handlers (lsp-ht ("emmy/progressReport" #'ignore))))
 
+
+(defgroup lsp-lua-language-server nil
+  "LSP support for Lua, using Lua Language Server."
+  :group 'lsp-mode
+  :link '(url-link "https://github.com/sumneko/lua-language-server"))
+
+(defcustom lsp-lua-language-server-install-dir (concat lsp-server-install-dir
+                                                       "/lua-language-server/")
+  "Installation directory for Lua Language Server."
+  :group 'lsp-lua-language-server
+  :type 'directory)
+
+(defcustom lsp-lua-language-server-bin (concat lsp-lua-language-server-install-dir
+                                               "bin/Linux/lua-language-server")
+  "Location of Lua Language Server."
+  :group 'lsp-lua-language-server
+  :type 'file)
+
+(defcustom lsp-lua-language-server-main-location (concat lsp-lua-language-server-install-dir
+                                                         "main.lua")
+  "Location of Lua Language Server main.lua."
+  :group 'lsp-lua-language-server
+  :type 'file)
+
+(defcustom lsp-lua-language-server-args "-E"
+  "List of arguments given to Lua Language server."
+  :group 'lsp-lua-language-server
+  :type 'file)
```

Comment:
It is not file but IMO you should also change the args to be (repeat string) and make "-E" to be a list otherwise it won't be possible to pass multiple flags. "-E -Foo" does not work.

Reply:
done